src/server/ghService/import/importPackageUtil.ts

```diff
@@ -1,132 +1,75 @@
-import { TextDecoder } from "util";
-
 import fetch from "node-fetch";
+import { BatchImport, Item } from "../types/import/batchImportModel";
 
-import * as jsYaml from "js-yaml";
-import { ManifestFolderList } from "../types/import/manifestFolderListModel";
-
+import parseYaml from "../helpers/decodingHelper";
 
 const {
   GITHUB_TOKEN,
 } = process.env;
 
-const CONTENTS_BASE_URL = "https://api.github.com/repos/microsoft/winget-pkgs/contents";
-
-//! only call for initial import
-const getManifestFolderPaths = async (): Promise<string[]> => {
-  const manifestFolderList: Promise<ManifestFolderList[]> = await fetch(
-    `${CONTENTS_BASE_URL}/manifests`, {
-      headers: {
-        Authorization: `token ${GITHUB_TOKEN}`,
-      },
-    },
-  ).then((res) => res.json());
-
-  const manifestFolderPaths = await (await manifestFolderList).map(
-    (x) => x.path,
-  );
-
-  return manifestFolderPaths;
-};
-
-const getPackageFolderPaths = async (): Promise<string[]> => {
-  //! only use for inital bulk import
-  const manifestFolderPaths = await getManifestFolderPaths();
-
-  const packageFolders: ManifestFolderList[] = await Promise.all(
-    manifestFolderPaths.map((e) => fetch(`${CONTENTS_BASE_URL}/${e}`, {
-      headers: {
-        Authorization: `token ${GITHUB_TOKEN}`,
-      },
-    }).then((res) => res.json())),
-  );
-
-  const flatPackageFolderPaths: ManifestFolderList[] = packageFolders.flat(
-    packageFolders.length,
-  );
-  const packageFolderPaths = flatPackageFolderPaths.map((x) => x.path);
+const BASE_URL = "https://api.github.com/search/code?q=extension:yaml+repo:microsoft/winget-pkgs+path:/manifests";
+const TAKE = "100";
```

Comment:
can this be a number? wont need the parseInt later as well

Reply:
done